Source/interfaces/IPlayerInfo.h

```diff
@@ -23,60 +23,67 @@
-            enum { ID = ID_PLAYER_PROPERTIES_AUDIO };
-            virtual ~IAudioIterator() {}
+         enum AudioCodec : uint8_t {
+            AUDIO_UNDEFINED,
+            AUDIO_AAC,
+            AUDIO_AC3,
+            AUDIO_AC3_PLUS,
+            AUDIO_DTS,
+            AUDIO_MPEG1,
+            AUDIO_MPEG2,
+            AUDIO_MPEG3,
+            AUDIO_MPEG4,
+            AUDIO_OPUS,
+            AUDIO_VORBIS_OGG,
+            AUDIO_WAV
+        };
 
-            enum AudioCodec : uint8_t {
-                UNDEFINED = 0,
-                AUDIO_AAC,
-                AUDIO_AC3,
-                AUDIO_AC3_PLUS,
-                AUDIO_DTS,
-                AUDIO_MPEG1,
-                AUDIO_MPEG2,
-                AUDIO_MPEG3,
-                AUDIO_MPEG4,
-                AUDIO_OPUS,
-                AUDIO_VORBIS_OGG,
-                AUDIO_WAV
-            };
+        typedef RPC::IIteratorType<AudioCodec, ID_PLAYER_PROPERTIES_AUDIO> IAudioCodedIterator;
 
-            virtual bool IsValid() const = 0;
-            virtual bool Next() = 0;
-            virtual void Reset() = 0;
-            virtual AudioCodec Codec() const = 0;
+        enum VideoCodec : uint8_t {
+            VIDEO_UNDEFINED,
+            VIDEO_H263,
+            VIDEO_H264,
+            VIDEO_H265,
+            VIDEO_H265_10,
+            VIDEO_MPEG,
+            VIDEO_VP8,
+            VIDEO_VP9,
+            VIDEO_VP10
         };
 
-        struct EXTERNAL IVideoIterator : virtual public Core::IUnknown {
-            enum { ID = ID_PLAYER_PROPERTIES_VIDEO };
-            virtual ~IVideoIterator() {}
+        typedef RPC::IIteratorType<VideoCodec, ID_PLAYER_PROPERTIES_VIDEO> IVideoCodecIterator;
 
-            enum VideoCodec : uint8_t {
-                UNDEFINED = 0,
-                VIDEO_H263,
-                VIDEO_H264,
-                VIDEO_H265,
-                VIDEO_H265_10,
-                VIDEO_MPEG,
-                VIDEO_VP8,
-                VIDEO_VP9,
-                VIDEO_VP10
-            };
+        virtual uint32_t AudioCodec(IAudioCodedIterator& codec /* @out */) const;
+        virtual uint32_t* VideoCodec(IVideoCodecIterator& codec /* @out */) const;
 
-            virtual bool IsValid() const = 0;
-            virtual bool Next() = 0;
-            virtual void Reset() = 0;
-            virtual VideoCodec Codec() const = 0;
+        enum PlaybackResolution : uint8_t {
+            RESOLUTION_UNKNOWN,
+            RESOLUTION_480I,
+            RESOLUTION_480P,
+            RESOLUTION_576I,
+            RESOLUTION_576P,
+            RESOLUTION_720P,
+            RESOLUTION_1080I,
+            RESOLUTION_1080P,
+            RESOLUTION_2160P30,
+            RESOLUTION_2160P60
         };
 
-        virtual IAudioIterator* AudioCodec() const = 0;
-        virtual IVideoIterator* VideoCodec() const = 0;
+        // @property
+        // @brief Current Video playback resolution set from EPG
```

Comment:
The interface documentation is generated from the @brief yu add here. I think it is important to leave implenataion notes (how it works on ComCast STB where it is coming from, like EPG) out of the comments.
At the end the only think we do here is report the Resolution used by the player. 
How you set it is a different story and not part of this interface.